python_ta/checkers/condition_logic_checker.py

```diff
@@ -0,0 +1,102 @@
+"""
+Check for redundant If or While conditions in functions based on z3 constraints
+"""
+
```

Comment:
Include a `from __future__ import annotations`, otherwise the type annotation `z3.ExprRef` below will raise an error if `z3` is not installed